src/main/java/uk/gov/hmcts/darts/retention/service/impl/ApplyRetentionProcessorImpl.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.darts.retention.service.impl;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.common.entity.CaseRetentionEntity;
+import uk.gov.hmcts.darts.common.repository.CaseRetentionRepository;
+import uk.gov.hmcts.darts.retention.enums.CaseRetentionStatus;
+import uk.gov.hmcts.darts.retention.service.ApplyRetentionProcessor;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@Slf4j
+@AllArgsConstructor
+public class ApplyRetentionProcessorImpl implements ApplyRetentionProcessor {
+
+    CaseRetentionRepository caseRetentionRepository;
+
+    @Override
+    public void processApplyRetention() {
+        List<CaseRetentionEntity> caseRetentionEntities =
+                caseRetentionRepository.findPendingRetention(OffsetDateTime.now().minusDays(7));
+        processList(caseRetentionEntities);
+
+    }
+
+    protected void processList(List<CaseRetentionEntity> caseRetentionEntities) {
+        List<Integer> processedCases = new ArrayList<>();
+        for (CaseRetentionEntity caseRetentionEntity: caseRetentionEntities) {
+            if (processedCases.contains(caseRetentionEntity.getId())) {
```

Comment:
should this be checking the courtCaseId instead of caseRetentionId?
Instead of skipping, would you want to update the status to "IGNORED"?
https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1700660840#RetentionServiceLLD-Processmapforretentionevents

Otherwise it will get picked up again in the next run?